orchagent/routeorch.cpp

```diff
@@ -649,6 +659,16 @@ bool RouteOrch::addRoute(IpPrefix ipPrefix, IpAddresses nextHops)
 
     m_syncdRoutes[ipPrefix] = nextHops;
 
+    /**
+     * If a route is added to m_syncdRoutes and was added to m_syncdTempRoutes,
+     * remove this entry from the m_syncdTempRoutes.
+     */
+    if (m_syncdTempRoutes.find(ipPrefix) != m_syncdTempRoutes.end() &&
+        m_syncdTempRoutes[ipPrefix] == nextHops)
+    {
+        m_syncdTempRoutes.erase(ipPrefix);
+    }
```

Comment:
what about remove route, do you remove the temproute is the route is removed before the next hop group is created?